core/src/main/java/ai/timefold/solver/core/impl/domain/solution/descriptor/InnerVariableMetaModel.java

```diff
@@ -0,0 +1,10 @@
+package ai.timefold.solver.core.impl.domain.solution.descriptor;
+
+import ai.timefold.solver.core.impl.domain.variable.descriptor.VariableDescriptor;
+
+public sealed interface InnerVariableMetaModel<Solution_>
```

Comment:
Why not move this interface to the public API and return a generic type? Alternatively, we could add the method to `VariableMetaModel`, as it appears to be a common method for all its child classes.

Reply:
The public API has only the methods it needs, and no more. Variable descriptors will not be exposed as public API - they do far too much, and expose far too many internals of the solver. In fact, the reason for metamodel's existence is so that I can hide the descriptors.